### PR TITLE
Recognize function bodies

### PR DESCRIPTION
**Pull request by pchaigno**
With this pull request, function and method bodies will be recognized under the scopes `meta.function.js` and `meta.method.js` with the headers.
This is a necessary step to then recognize objects.

### CODE REVIEW

**Comment by pchaigno** [comment]
@50Wliu Travis build fixed!

**Comment by MaximSokolov** [comment]
Hmm. All languages use `meta.function` scope only for function declaration. [TextMate manual](http://manual.macromates.com/en/language_grammars#naming_conventions) says that `meta.function` match function declaration:
> ... For example the entire line which declares a function would be `meta.function`...

I doubt we should change it.

I [proposed](https://github.com/atom/docs/pull/114#issuecomment-136132055) `meta.function-call` and `meta.method-call` scopes expecting that `meta.function` would match functions' declarations. So if we merge this PR this code:
```Less
.meta.function {
  .entity.name.function {
      color: @color1;
  }
}
``` 
... would also highlight functions/methods calls:
```js
function() {
  // `functionCall` would be highlighted
  // I assume `functionCall` is `entity.name.function`
  functionCall();
}
```
I suggest to add new scope under `meta.*` for functions' bodies.

**Comment by 50Wliu** [comment]
Can we just make `function-params` also have the => part?  Seems like a win-win.  In addition, `( x =>` doesn't seem valid to me.  `x =>` or `(x) =>` should be the only two that are valid, but I'm not exactly a Javascript expert.

**Comment by MaximSokolov** [comment]
How about `declaration` sub-scope?
```less
.function.declaration {
  .entity.name {
      color: @color1;
  }
}

.function-call, .method-call {
  .entity.name {
      color: @color2;
  }
}

```

File: grammars/javascript.cson [hunk 1]
```diff
@@ -76,18 +76,19 @@
...
-        'name': 'punctuation.definition.parameters.begin.js'
     'comment': 'match stuff like: Sound.prototype.play = function() { … }'
-    'end': '(\\))'
+    'end': '\\}'
     'endCaptures':
-      '1':
-        'name': 'punctuation.definition.parameters.end.js'
+      '0':
+        'name': 'punctuation.definition.function.body.end.js'
     'name': 'meta.function.prototype.js'
```
Comment (50Wliu) [inline grammars/javascript.cson]: I think this needs to be updated, because the entire function block shouldn't be tokenized as a prototype.

**Comment by 50Wliu** [comment]
@pchaigno A few comments:
1. The `name`s of each have to be updated I think.  Maybe just move it to capture `0` of `beginCaptures` or give it a new name.
2. Can `function-params` be included in `function-body`?
3. There's conflicts :(

**Pull request closed without merging**